lib/apiservers/service/restapi/handlers/vch_log_get.go

```diff
@@ -0,0 +1,200 @@
+	}
+
+	log.Printf("done building data")
+
+	d.ID = params.VchID
+	op := trace.NewOperation(params.HTTPRequest.Context(), "vch: %s", params.VchID)
+
+	helper, err := getDatastoreHelper(op.Context, d)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	logFilePaths, err := getAllLogFilePaths(op.Context, helper)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	output, err := getContentFromLogFile(op.Context, helper, logFilePaths)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	return operations.NewGetTargetTargetVchVchIDLogOK().WithPayload(output)
+}
+
+func (h *VCHDatacenterLogGet) Handle(params operations.GetTargetTargetDatacenterDatacenterVchVchIDLogParams, principal interface{}) middleware.Responder {
+	d, err := buildData(params.HTTPRequest.Context(),
+		url.URL{Host: params.Target},
+		principal.(Credentials).user,
+		principal.(Credentials).pass,
+		params.Thumbprint,
+		&params.Datacenter,
+		nil)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	d.ID = params.VchID
+	op := trace.NewOperation(params.HTTPRequest.Context(), "vch: %s", params.VchID)
+
+	helper, err := getDatastoreHelper(op.Context, d)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	logFilePaths, err := getAllLogFilePaths(op.Context, helper)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	output, err := getContentFromLogFile(op.Context, helper, logFilePaths)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogOK().WithPayload(output)
+}
+
+// getDatastoreHelper validates the VCH and returns the datastore helper for the VCH. It errors when validation fails or when datastore is not ready
+func getDatastoreHelper(ctx context.Context, d *data.Data) (*datastore.Helper, error) {
+	// TODO (angiew): abstract some of the boilerplate into helpers in common.go
+	validator, err := validateTarget(ctx, d)
+	if err != nil {
+		return nil, util.WrapError(400, err)
+	}
+
+	executor := management.NewDispatcher(validator.Context, validator.Session, nil, false)
+	vch, err := executor.NewVCHFromID(d.ID)
+	if err != nil {
+		return nil, util.NewError(500, fmt.Sprintf("failed to create VCH %s: %s", d.ID, err))
```

Comment:
"failed to create VCH" doesn't seem like quite the right message

Reply:
This piece is copied from `getVCHConfig` function in `common.go`
I'll update both places.
What should be a more suitable message? like `VCH not found`?